src/compiler/moduleSpecifiers.ts

```diff
@@ -271,37 +310,14 @@ namespace ts.moduleSpecifiers {
         return removeFileExtension(relativePath);
     }
 
-    function tryGetModuleNameFromTypeRoots(

```

Comment:
See below comment about `typeRoots`. `node_modules/@types` is always used for module resolution even if it's not in typeRoots; that's now handled in `tryGetModuleNameAsNodeModule`.